devtools/cmd/migrate-librarian/main.go

```diff
@@ -0,0 +1,202 @@
+	errTidyFailed       = errors.New("librarian tidy failed")
+)
+
+func main() {
+	if err := run(os.Args[1:]); err != nil {
+		slog.Error("migrate-librarian failed", "err", err)
+		os.Exit(1)
+	}
+}
+
+func run(args []string) error {
+	flagSet := flag.NewFlagSet("migrate-librarian", flag.ContinueOnError)
+	repoPath := flagSet.String("repo", "", "Path to the repository containing legacy .librarian configuration (required)")
+	language := flagSet.String("lang", "go", "One of go and python (default: go)")
+	outputPath := flagSet.String("output", "./librarian.yaml", "Output file path (default: ./librarian.yaml)")
+	if err := flagSet.Parse(args); err != nil {
+		return err
+	}
+	if *repoPath == "" {
+		return errRepoNotFound
+	}
+
+	if *language != "go" && *language != "python" {
+		return errLangNotSupported
+	}
+
+	librarianState, err := readState(*repoPath)
+	if err != nil {
+		return err
+	}
+
+	librarianConfig, err := readConfig(*repoPath)
+	if err != nil {
+		return err
+	}
+
+	cfg := buildConfig(librarianState, librarianConfig, *language)
+
+	if err := yaml.Write(*outputPath, cfg); err != nil {
+		return fmt.Errorf("failed to write config: %w", err)
+	}
+	slog.Info("Wrote config to output file", "path", outputPath)
+
+	if err := librarian.RunTidy(); err != nil {
+		slog.Error(errTidyFailed.Error(), "error", err)
+		return errTidyFailed
+	}
+
+	return nil
+}
+
+func buildConfig(
+	librarianState *legacyconfig.LibrarianState,
+	librarianConfig *legacyconfig.LibrarianConfig,
+	lang string) *config.Config {
+	repo := "googleapis/google-cloud-go"
+	if lang == "python" {
+		repo = "googleapis/google-cloud-python"
+	}
+
+	cfg := &config.Config{
+		Language: lang,
+		Repo:     repo,
+		Default: &config.Default{
+			TagFormat: defaultTagFormat,
+		},
+	}
+
+	libraries, sha := buildLibraries(librarianState, librarianConfig)
+	if sha != "" {
+		cfg.Sources = &config.Sources{
+			Googleapis: &config.Source{
+				Commit: sha,
+			},
+		}
+	}
+	cfg.Libraries = libraries
+
+	return cfg
+}
+
+func buildLibraries(
+	librarianState *legacyconfig.LibrarianState,
+	librarianConfig *legacyconfig.LibrarianConfig) ([]*config.Library, string) {
+	var libraries []*config.Library
+	idToLibraryState := sliceToMap[legacyconfig.LibraryState](
+		librarianState.Libraries,
+		func(lib *legacyconfig.LibraryState) string {
+			return lib.ID
+		})
+	idToLibraryConfig := sliceToMap[legacyconfig.LibraryConfig](
+		librarianConfig.Libraries,
+		func(lib *legacyconfig.LibraryConfig) string {
+			return lib.LibraryID
+		})
+	googleapisCommitSHA := ""
+	// Iterate libraries from idToLibraryState because librarianConfig.Libraries is a
+	// subset of librarianState.Libraries.
+	for id, libState := range idToLibraryState {
+		library := &config.Library{}
+		library.Name = id
+		library.Version = libState.Version
+		if libState.APIs != nil {
+			library.Channels = toChannels(libState.APIs)
+		}
+		library.Keep = libState.PreserveRegex
```

Comment:
I don't think we need all of the keep lines. For example, for Go, it looks like this:

```
    keep:
      - packages/google-ads-admanager/CHANGELOG.md
      - docs/CHANGELOG.md
      - docs/README.rst
      - samples/README.txt
      - scripts/client-post-processing
      - samples/snippets/README.rst
      - tests/system
 ```

I think these lines appear for every single library, in which case we should just hardcode it into the generator as opposed to list it for all 150+ libraries.

It would be helpful to dedupe (feel free to do as a follow up)

Reply:
I didn't see non-empty preserve regex in https://github.com/googleapis/google-cloud-go/blob/main/.librarian/state.yaml